// naive-ui/lib/_utils/naive/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.warnOnce = exports.warn = exports.throwError = exports.isArrayShallowEqual = exports.smallerSize = exports.largerSize = exports.getTitleAttribute = void 0;
var attribute_1 = require("./attribute");
Object.defineProperty(exports, "getTitleAttribute", { enumerable: true, get: function () { return attribute_1.getTitleAttribute; } });
var prop_1 = require("./prop");
Object.defineProperty(exports, "largerSize", { enumerable: true, get: function () { return prop_1.largerSize; } });
Object.defineProperty(exports, "smallerSize", { enumerable: true, get: function () { return prop_1.smallerSize; } });
var value_1 = require("./value");
Object.defineProperty(exports, "isArrayShallowEqual", { enumerable: true, get: function () { return value_1.isArrayShallowEqual; } });
var warn_1 = require("./warn");
Object.defineProperty(exports, "throwError", { enumerable: true, get: function () { return warn_1.throwError; } });
Object.defineProperty(exports, "warn", { enumerable: true, get: function () { return warn_1.warn; } });
Object.defineProperty(exports, "warnOnce", { enumerable: true, get: function () { return warn_1.warnOnce; } });
