// naive-ui/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useOsTheme = exports.zindexable = exports.version = exports.lightTheme = exports.darkTheme = exports.createTheme = exports.NThemeEditor = exports.install = exports.default = exports.create = exports.cNotM = exports.cM = exports.cE = exports.cB = exports.c = void 0;
var cssr_1 = require("./_utils/cssr");
Object.defineProperty(exports, "c", { enumerable: true, get: function () { return cssr_1.c; } });
Object.defineProperty(exports, "cB", { enumerable: true, get: function () { return cssr_1.cB; } });
Object.defineProperty(exports, "cE", { enumerable: true, get: function () { return cssr_1.cE; } });
Object.defineProperty(exports, "cM", { enumerable: true, get: function () { return cssr_1.cM; } });
Object.defineProperty(exports, "cNotM", { enumerable: true, get: function () { return cssr_1.cNotM; } });
__exportStar(require("./components"), exports);
__exportStar(require("./composables"), exports);
var create_1 = require("./create");
Object.defineProperty(exports, "create", { enumerable: true, get: function () { return __importDefault(create_1).default; } });
__exportStar(require("./locales"), exports);
var preset_1 = require("./preset");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return __importDefault(preset_1).default; } });
Object.defineProperty(exports, "install", { enumerable: true, get: function () { return preset_1.install; } });
// component themes
__exportStar(require("./styles"), exports);
var theme_editor_1 = require("./theme-editor");
Object.defineProperty(exports, "NThemeEditor", { enumerable: true, get: function () { return theme_editor_1.NThemeEditor; } });
// composed global theme, createTheme from component themes util
var themes_1 = require("./themes");
Object.defineProperty(exports, "createTheme", { enumerable: true, get: function () { return themes_1.createTheme; } });
Object.defineProperty(exports, "darkTheme", { enumerable: true, get: function () { return themes_1.darkTheme; } });
Object.defineProperty(exports, "lightTheme", { enumerable: true, get: function () { return themes_1.lightTheme; } });
var version_1 = require("./version");
Object.defineProperty(exports, "version", { enumerable: true, get: function () { return __importDefault(version_1).default; } });
var vdirs_1 = require("vdirs");
Object.defineProperty(exports, "zindexable", { enumerable: true, get: function () { return vdirs_1.zindexable; } });
var vooks_1 = require("vooks");
Object.defineProperty(exports, "useOsTheme", { enumerable: true, get: function () { return vooks_1.useOsTheme; } });
