// naive-ui/lib/dialog/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NDialogProvider = exports.dialogProviderProps = exports.dialogProps = exports.NDialog = exports.useDialogReactiveList = exports.useDialog = void 0;
var composables_1 = require("./src/composables");
Object.defineProperty(exports, "useDialog", { enumerable: true, get: function () { return composables_1.useDialog; } });
Object.defineProperty(exports, "useDialogReactiveList", { enumerable: true, get: function () { return composables_1.useDialogReactiveList; } });
var Dialog_1 = require("./src/Dialog");
Object.defineProperty(exports, "NDialog", { enumerable: true, get: function () { return Dialog_1.NDialog; } });
var dialogProps_1 = require("./src/dialogProps");
Object.defineProperty(exports, "dialogProps", { enumerable: true, get: function () { return dialogProps_1.dialogProps; } });
var DialogProvider_1 = require("./src/DialogProvider");
Object.defineProperty(exports, "dialogProviderProps", { enumerable: true, get: function () { return DialogProvider_1.dialogProviderProps; } });
Object.defineProperty(exports, "NDialogProvider", { enumerable: true, get: function () { return DialogProvider_1.NDialogProvider; } });
