// naive-ui/lib/_mixins/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useTheme = exports.createTheme = exports.useStyle = exports.useRtl = exports.useLocale = exports.useHljs = exports.useFormItem = exports.useThemeClass = exports.useConfig = exports.defaultClsPrefix = void 0;
var use_config_1 = require("./use-config");
Object.defineProperty(exports, "defaultClsPrefix", { enumerable: true, get: function () { return use_config_1.defaultClsPrefix; } });
Object.defineProperty(exports, "useConfig", { enumerable: true, get: function () { return __importDefault(use_config_1).default; } });
var use_css_vars_class_1 = require("./use-css-vars-class");
Object.defineProperty(exports, "useThemeClass", { enumerable: true, get: function () { return use_css_vars_class_1.useThemeClass; } });
var use_form_item_1 = require("./use-form-item");
Object.defineProperty(exports, "useFormItem", { enumerable: true, get: function () { return __importDefault(use_form_item_1).default; } });
var use_hljs_1 = require("./use-hljs");
Object.defineProperty(exports, "useHljs", { enumerable: true, get: function () { return __importDefault(use_hljs_1).default; } });
var use_locale_1 = require("./use-locale");
Object.defineProperty(exports, "useLocale", { enumerable: true, get: function () { return __importDefault(use_locale_1).default; } });
var use_rtl_1 = require("./use-rtl");
Object.defineProperty(exports, "useRtl", { enumerable: true, get: function () { return use_rtl_1.useRtl; } });
var use_style_1 = require("./use-style");
Object.defineProperty(exports, "useStyle", { enumerable: true, get: function () { return __importDefault(use_style_1).default; } });
var use_theme_1 = require("./use-theme");
Object.defineProperty(exports, "createTheme", { enumerable: true, get: function () { return use_theme_1.createTheme; } });
Object.defineProperty(exports, "useTheme", { enumerable: true, get: function () { return __importDefault(use_theme_1).default; } });
